src/internal/connector/sharepoint/list.go

```diff
@@ -0,0 +1,452 @@
+			cont.SetColumnLinks(links)
+
+			positions, err := fetchColumnPositions(ctx, gs, identifier, id)
+			if err != nil {
+				errs = support.WrapAndAppend("unable to add column definitionable to list", err, errs)
+				break
+			}
+
+			cont.SetColumnPositions(positions)
+
+			cs, err := fetchContentColumns(ctx,
+				gs,
+				identifier,
+				gs.Client().SitesById(identifier).ContentTypesById(*cont.GetId()).Columns())
+			if err != nil {
+				errs = support.WrapAndAppend("unable to populate columns for contentType", err, errs)
+			}
+
+			cont.SetColumns(cs)
+
+			bTypes, err := fetchContentBaseTypes(ctx, gs, identifier, id)
+			if err != nil {
+				errs = support.WrapAndAppend("unable to add baseTypes to List", err, errs)
+				break
+			}
+
+			cont.SetBaseTypes(bTypes)
+
+			cTypes = append(cTypes, cont)
+		}
+
+		if resp.GetOdataNextLink() == nil {
+			break
+		}
+
+		builder = contenttypes.NewContentTypesRequestBuilder(*resp.GetOdataNextLink(), gs.Adapter())
+	}
+
+	if errs != nil {
+		return nil, errs
+	}
+
+	return cTypes, nil
+}
+
+func fetchContentBaseTypes(
+	ctx context.Context,
+	gs graph.Service,
+	siteID, cTypeID string,
+) ([]models.ContentTypeable, error) {
+	var (
+		errs    error
+		cTypes  = make([]models.ContentTypeable, 0)
+		builder = gs.Client().SitesById(siteID).ContentTypesById(cTypeID).BaseTypes()
+	)
+
+	for {
+		resp, err := builder.Get(ctx, nil)
+		if err != nil {
+			return nil, errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+		}
+
+		for _, obj := range resp.GetValue() {
+			cBuilder := gs.Client().SitesById(siteID).ContentTypesById(*obj.GetId()).Columns()
+
+			cs, err := fetchContentColumns(ctx, gs, siteID, cBuilder)
+			if err != nil {
+				errs = support.WrapAndAppend("columns not added on fetch baseType", err, errs)
+			}
+
+			obj.SetColumns(cs)
+
+			lnk, err := fetchColumnLinks(ctx, gs, siteID, *obj.GetId())
+			if err != nil {
+				errs = support.WrapAndAppend("columnLink failure on fetch baseType", err, errs)
+			}
+
+			obj.SetColumnLinks(lnk)
+
+			pos, err := fetchColumnPositions(ctx, gs, siteID, *obj.GetId())
+			if err != nil {
+				errs = support.WrapAndAppend("column position not added on fetch baseType", err, errs)
+			}
+
+			obj.SetColumnPositions(pos)
+
+			cTypes = append(cTypes, obj)
+		}
+
+		if resp.GetOdataNextLink() == nil {
+			break
+		}
+
+		builder = basetypes.NewBaseTypesRequestBuilder(*resp.GetOdataNextLink(), gs.Adapter())
+	}
+
+	if errs != nil {
+		return nil, errs
+	}
+
+	return cTypes, nil
+}
+
+func fetchColumnLinks(ctx context.Context, gs graph.Service, siteID, cTypeID string) ([]models.ColumnLinkable, error) {
+	var (
+		builder = gs.Client().SitesById(siteID).ContentTypesById(cTypeID).ColumnLinks()
+		links   = make([]models.ColumnLinkable, 0)
+	)
+
+	for {
+		resp, err := builder.Get(ctx, nil)
+		if err != nil {
+			return nil, errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+		}
+
+		for _, link := range resp.GetValue() {
+			links = append(links, link)
+		}
```

Comment:
Fun tip: you can append slices without iteration by using a spread operator:
```suggestion
		links = append(links, resp.GetValue()...)
```

Reply:
Resolved as of [6e7fdb2](https://github.com/alcionai/corso/pull/1633/commits/6e7fdb245276f7ff194e8d6e1e37f79110246330)